audio/quarto_mexca_performance_diarization.qmd

```diff
@@ -0,0 +1,378 @@
+
+  }
+}
+```
+
+```{r}
+head(F1_total)
+```
+
+## F1 score distribution
+
+F1 score as a function of the number of speakers. Each dot is a speaker with its own speaking time duration (\~40, \~60, \~120, \~180, \~300, \~480, \~600). F1 scores close to 0 imply perfect confusion, F1 scores close to 1 imply perfect identification.
+
+```{r, warning=FALSE}
+
+ggplot(F1_total, aes(as.factor(n_speakers), f1_score))+
+  geom_violin(fill = "lightgray", col = NA)+
+  geom_sina(shape = 21, size = 2)+
+  scale_y_continuous(breaks=seq(0, 1, 0.2), limits=c(0, 1))+
+  ylab("f1 score")+
+  xlab("number of people talking in total")+
+  theme_classic()+
+  stat_summary(fun.data="mean_cl_boot", color="red")
+```
+
+We can see that the performance is very high, and it looks constant across `n` except for some instances (10 speakers/datapoints).
+
+Let's inspect more closely each speaker:
+
+```{r, warning=FALSE}
+ggplot(F1_total, aes(as.factor(speakers), f1_score)) +
+  geom_violin(fill = "lightgray", col = NA) +
+  geom_sina(shape = 21, size = 2) +
+  ylab("f1 score") +
+  xlab("speaker") +
+  theme_classic() +
+  stat_summary(fun.data="mean_cl_boot", color="red")
+
+```
+
+It is evident that the poor performance in those datapoints can be easily attributed to two speakers (Klaver and Wilders) and for 1 instance in Burger. Apart from Burger (that it might be just bad luck), this plot suggests that f1_score seems to be influenced more by the singular features of the speakers rather than the total number of speakers.
+
+Let's look now at whether speaking time has any influence:
+
+```{r}
+ggplot(F1_total, aes(as.factor(file), f1_score)) +
+  geom_violin(fill = "lightgray", col = NA) +
+  geom_sina(shape = 21, size = 2) +
+  ylab("f1 score") +
+  xlab("time(s)") +
+  theme_classic() +
+  stat_summary(fun.data="mean_cl_boot", color="red")
+```
+
+F1 score decreases at longer speaking time.
+
+## Confusion matrices
+
+```{r, figures-side, fig.ncol = 3, out.height="300px"}
+
+ for(i in 1:length(n_speaker)){
+   for(x in 1:length(max_duration)){
+     print(my2Dlist[[i,x]])
+   }
+ }
+
+```
+
+## Q1: Is performance affected by the number of speakers in the audio?
+
+`Answer:` No. Performance is constant across speakers (i.e., it is not influenced by n). When performance breaks down it doesn't do it uniformly, the model occasionally fails to recognise particular speakers (e.g., Klaver) while the rest remain well classified.
+
+## Q2: Is performance influenced by how long one speak compared to the others?
+
+`Answer`: Yes. Performance decreases for speakers whose speaking time is over \~8 min.
+
+`Conclusion:` Our results suggest that the audio pipeline can be safely used for any number of speaker (up to the tested number of 7) without decreasing in performance, however seems likely to be affected by how long one speaks compared to the others. To be sure though, one needs to analyse also a more balanced dataset (e.g., whereby each speaker speaks the same amount of time).
```

Comment:
I would not use the word *safely* here because the classifier completely misidentifies some speakers in (admittedly) rare cases.

Reply:
I agree with your wording. I am going to change that also in the video notebook